packages/next-auth/src/next/index.ts

```diff
@@ -83,25 +83,31 @@ function NextAuth(
 export default NextAuth
 
 export async function getServerSession(
-  context:
-    | GetServerSidePropsContext
-    | { req: NextApiRequest; res: NextApiResponse },
-  options: NextAuthOptions
+  ...args:
+    | [GetServerSidePropsContext['req'], GetServerSidePropsContext['res'], NextAuthOptions]
+    | [NextApiRequest, NextApiResponse, NextAuthOptions]
 ): Promise<Session | null> {
```

Comment:
I wonder if we should return this instead:

```ts
interface GetServerSessionResult {
  session: Promise<Session | null>
  redirect?: string
}
```

And do this similar to the `authorized` callback in `next-auth/middleware`: https://next-auth.js.org/configuration/nextjs#callbacks

We should then add the `callbacks.authorized` logic inside `getServerSession` as well. (should we just alias `callbacks.signIn`? #2855)

I reckon it's a common requirement to redirect to a sign-in page if the session was not found.

What do you think?

Reply:
While it's common to redirect to a sign-in page, I feel like we shouldn't limit users to that option only. It should be easy for them to override this behavior when they want 🤔